remote-api/src/main/java/org/alfresco/repo/web/scripts/content/ContentGet.java

```diff
@@ -140,6 +148,22 @@ public void execute(WebScriptRequest req, WebScriptResponse res)
                 propertyQName = QName.createQName(propertyName, namespaceService);
             }
         }
+        // determine attachment and force download for specific mimetypes - see PRODSEC-5862
+        boolean attach = Boolean.valueOf(req.getParameter("a"));
+        ContentReader reader = contentService.getReader(nodeRef, propertyQName);
+        String mimetype = MimeTypeUtil.determineMimetype(reader,req,mimetypeService);
+
+        if (!attach)
+        {
+            if ((nonAttachContentTypes != null) && (!(nonAttachContentTypes.contains(mimetype))))
```

Comment:
This conditional has not been inverted properly. If it was `nonAttachContentTypes != null && nonAttachContentTypes.contains(mimetype)`, then the negation should be `nonAttachContentTypes == null || !nonAttachContentTypes.contains(mimetype)`.

Also, there are a lot of unnecessary `(` and `)` characters. Let's get rid of them.

Reply:
nonAttachContentTypes is just a check to see if there are any content types restricted for showing in browser, as it was also done this way in other places (I guess in case we ever change our mind about that). If you think its not necessary I can remove it completely.  It should not be inverted together with the other check which actually is part of the logic of the operation, as opposed to just checking if the list of acceptable mime types is empty

Reply:
Clarified via quick Zoom meeting that we do need to completely negate the conditional clause.